### PR TITLE
Add the next button navigation view to a default long scroll

### LINKED ISSUE #1764: Only enable next button when user scrolls to the bottom of the page
**Is your feature request related to a problem? Please describe.**
Sometimes there is a display item that is enabled at the bottom of the page that a user needs to see before they proceed to the next page.

**Describe the solution you'd like**
The next button can be disabled by default and only enabled once the user scrolls all the way to the bottom of the page.

**Describe alternatives you've considered**
I can't think of anything else that prompts the user to scroll to view more information. If there is something, please let me know.

**Additional context**
As shown in the video below, the user can proceed to the next page without seeing the prompt "viral load collection is due today"
https://user-images.githubusercontent.com/90775548/207011675-8d3da458-2bf4-48ca-9198-92296f000bf6.mp4

**Would you like to work on the issue?**
N/A

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1764 

**Description**
Adds support for having the bottom navigation buttons as part of the Questionnaire recyclerview, that would ensure the user has viewed all the field items before clicking next 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

Detecting scroll to the bottom of the Recyclerview to toggle visibility of the bottom buttons - was not smooth and the bottom buttons were not showing in a predictable manner

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by LZRS** [comment]
[Screen_recording_20231019_122501.webm](https://github.com/google/android-fhir/assets/12814349/594dcd45-8199-4036-9f3b-b40dedef8937)

**Comment by LZRS** [comment]
screen_recording

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -663,9 +700,49 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
...
+            QuestionnaireNavigationViewState.Hidden
+          },
+        nextPageNavigationActionState =
+          if (questionnairePagination.isPaginated && questionnairePagination.hasNextPage) {
+            QuestionnaireNavigationViewState.Enabled { goToNextPage() }
+          } else if (questionnairePagination.isPaginated) {
+            QuestionnaireNavigationViewState.Disabled
+          } else {
+            QuestionnaireNavigationViewState.Hidden
+          },
```
Comment (santosh-pingle) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt:720]: ditto

**Comment by MJ1998** [comment]
@santosh-pingle Since you are looking into this PR, its ready for review.
@LZRS Requesting to confirm and check all the items under checklist in the issue description.

**Comment by jingtang10** [comment]
@LZRS thanks for this PR - it's a very well written PR!

I want to highlight though in @shelaghm's comment here https://github.com/google/android-fhir/issues/1764#issuecomment-1433889489 she said `My concern with option 1 is hiding/disabling buttons can also lead to user confusion, and I have a preference for keeping paginated navigation buttons anchored and always visible. We have some [design guidelines](https://developers.google.com/open-health-stack/design/data-capture-guideline#navigation_buttons) here with the rationale`.

Your PR changes this right? is there an option to keep the current behaviour? if so can you add a video?

**Comment by jingtang10** [comment]
looks good - can you please add tests @LZRS ?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireAdapterItem.kt [hunk 2]
```diff
@@ -22,4 +22,7 @@ import com.google.android.fhir.datacapture.views.QuestionnaireViewItem
 internal sealed interface QuestionnaireAdapterItem {
   /** A row for a question in a Questionnaire RecyclerView. */
   data class Question(val item: QuestionnaireViewItem) : QuestionnaireAdapterItem
+
+  data class Navigation(val questionnaireNavigationUIState: QuestionnaireNavigationUIState) :
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireAdapterItem.kt:26]: @kevinmost fyi - this proves to be useful :)

**Review by jingtang10 — APPROVED** [review]
Thanks @LZRS again for this. I've updated this PR and done a couple of improvements:

1. removed the unnecessary navigation recycler view. in the case where the navigation controls are floating at the bottom of the screen, they can just be a simple reused layout instead of using a recycler view which is costly.
2. removed the disabled state for pagination controls which I don't think is ever used actually.

I also want to simplify the QuestionnaireNavigationViewUIState class altogether as I find it to be overly complicated.

I'll try to merge this PR asap.

Thanks

**Pull request merged**